### PR TITLE
feature (refs T31435): delete boilerplates of boilerplateGroup on GroupDelete

### PR DESCRIPTION
**Pull request by MoritzMandler**
Ticket:
https://yaits.demos-deutschland.de/T31435

Description:
delete boilerplates of boilerplateGroups when a boilerplate group
is selected to be deleted.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php [hunk 1]
```diff
@@ -2781,11 +2809,25 @@ protected function handleDeleteBoilerplate(string $boilerplateId)
...
+    ) {
+        $boilerplatesOfGroupToDelete = new ArrayCollection();
         $boilerplateGroupToDelete = $this->procedureService->getBoilerplateGroup($boilerplateGroupId);
+        if (null !== $boilerplateGroupToDelete) {
+            foreach ($boilerplateGroupToDelete->getBoilerplates() as $boilerplate) {
+                $boilerplatesOfGroupToDelete->add($boilerplate);
+            }
+        }
+
         $title = null === $boilerplateGroupToDelete ? '' : $boilerplateGroupToDelete->getTitle();
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php:2824]: I think that $successfully is false when $boilerplateGroupToDelete is null. For this reason I think we can remove the $title here and can $boilerplateGroupToDelete->getTitle() move to the MessageBag directly.

**Review by breierDemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php (hunk 1 above)
Comment (breierDemos) [reply demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php:2824]: I'm wrong we delete $boilerplateGroupToDelete before we add the message to the MessageBag so it is null.

File: demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php [hunk 2]
```diff
@@ -2781,11 +2809,25 @@ protected function handleDeleteBoilerplate(string $boilerplateId)
...
+
         $successfully = $this->procedureService->deleteBoilerplateGroup($boilerplateGroupToDelete);
+        /** @var Boilerplate $boilerplate */
+        foreach ($boilerplatesOfGroupToDelete as $boilerplate) {
+            $successfully = $successfully && $boilerplateRepository->delete($boilerplate->getId());
```
[the remark this answers is not in the document]
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php]: done in 3c6f664e27bc8d54e54618ef714a39dc0577b6f4

File: demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php [hunk 3]
```diff
@@ -2819,9 +2861,26 @@ protected function handleDeleteBoilerplates(
...
+                continue;
+            }
+            foreach ($boilerplateGroup->getBoilerplates() as $boilerplate) {
+                $boilerplatesOfGroupsToDelete->add($boilerplate);
+            }
+        }
         $allDeleted = $this->procedureService->deleteBoilerplateGroupsByIds($boilerplateGroupIds);
+        /** @var Boilerplate $boilerplate */
+        foreach ($boilerplatesOfGroupsToDelete as $boilerplate) {
+            $allDeleted = $allDeleted && $boilerplateRepository->delete($boilerplate->getId());
```
[the remark this answers is not in the document]
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Controller/Procedure/DemosPlanProcedureController.php]: done in 3c6f664e27bc8d54e54618ef714a39dc0577b6f4

**Review by breierDemos — APPROVED** [review]
lgtm

**Pull request merged**